Godot from master branch don't build in Gentoo

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
<!-- Specify commit hash if using non-official build. -->
ba51af7904819b3739173d24567e79ee5d3600f9

**OS/device including version:**
<!-- Specify GPU model, drivers, and the backend (GLES2, GLES3, Vulkan) if graphics-related. -->
Gentoo Linux AMD64

**Issue description:**
<!-- What happened, and what was expected. -->
```
[Initial build] Compiling ==> platform/linuxbsd/display_server_x11.cpp
platform/linuxbsd/display_server_x11.cpp:809:2: warning: #warning Need to get from proper window [-Wcpp]
  809 | #warning Need to get from proper window
      |  ^~~~~~~
platform/linuxbsd/display_server_x11.cpp:3990:2: warning: #warning Forcing vulkan rendering driver because OpenGL not implemented yet [-Wcpp]
 3990 | #warning Forcing vulkan rendering driver because OpenGL not implemented yet
      |  ^~~~~~~
platform/linuxbsd/display_server_x11.cpp: In member function 'String DisplayServerX11::_clipboard_get_impl(Atom, Window, Atom) const':
platform/linuxbsd/display_server_x11.cpp:541:11: error: 'LONG_MAX' was not declared in this scope
  541 |        0, LONG_MAX, // offset - len
      |           ^~~~~~~~
platform/linuxbsd/display_server_x11.cpp:76:1: note: 'LONG_MAX' is defined in header '<climits>'; did you forget to '#include <climits>'?
   75 | #include <X11/XKBlib.h>
  +++ |+#include <climits>
   76 | 
platform/linuxbsd/display_server_x11.cpp: In member function 'void DisplayServerX11::_handle_selection_request_event(XSelectionRequestEvent*) const':
platform/linuxbsd/display_server_x11.cpp:2501:81: error: 'LONG_MAX' was not declared in this scope
 2501 |   if (XGetWindowProperty(x11_display, p_event->requestor, p_event->property, 0, LONG_MAX, False, atom_pair, &type, &format, &len, &remaining, &data) == Success) {
      |                                                                                 ^~~~~~~~
platform/linuxbsd/display_server_x11.cpp:2501:81: note: 'LONG_MAX' is defined in header '<climits>'; did you forget to '#include <climits>'?
scons: *** [platform/linuxbsd/display_server_x11.linuxbsd.opt.tools.64.o] Error 1
scons: building terminated because of errors.
```

**Steps to reproduce:**

1. Checkout latest sources.
2. Build then on linux with `scons platform=linuxbsd target=release_debug`

Also happens in Ubuntu 20.10

    @  cpp/godot   master  scons platform=x11
    scons: Reading SConscript files ...
    Platform "x11" has been renamed to "linuxbsd" in Godot 4.0. Building for platform "linuxbsd".
    Enabling ALSA
    PulseAudio development libraries not found, disabling driver
    Checking for C header file mntent.h... (cached) yes
    scons: done reading SConscript files.
    scons: Building targets ...
    [  2%] Compiling ==> platform/linuxbsd/display_server_x11.cpp
    platform/linuxbsd/display_server_x11.cpp:809:2: warning: #warning Need to get from proper window [-Wcpp]
    809 | #warning Need to get from proper window
        |  ^~~~~~~
    platform/linuxbsd/display_server_x11.cpp:3990:2: warning: #warning Forcing vulkan rendering driver because OpenGL not implemented yet [-Wcpp]
    3990 | #warning Forcing vulkan rendering driver because OpenGL not implemented yet
        |  ^~~~~~~
    platform/linuxbsd/display_server_x11.cpp: In member function 'String DisplayServerX11::_clipboard_get_impl(Atom, Window, Atom) const':
    platform/linuxbsd/display_server_x11.cpp:541:11: error: 'LONG_MAX' was not declared in this scope
    541 |        0, LONG_MAX, // offset - len
        |           ^~~~~~~~
    platform/linuxbsd/display_server_x11.cpp:76:1: note: 'LONG_MAX' is defined in header '<climits>'; did you forget to '#include <climits>'?
    75 | #include <X11/XKBlib.h>
    +++ |+#include <climits>
    76 | 
    platform/linuxbsd/display_server_x11.cpp: In member function 'void DisplayServerX11::_handle_selection_request_event(XSelectionRequestEvent*) const':
    platform/linuxbsd/display_server_x11.cpp:2501:81: error: 'LONG_MAX' was not declared in this scope
    2501 |   if (XGetWindowProperty(x11_display, p_event->requestor, p_event->property, 0, LONG_MAX, False, atom_pair, &type, &format, &len, &remaining, &data) == Success) {
        |                                                                                 ^~~~~~~~
    platform/linuxbsd/display_server_x11.cpp:2501:81: note: 'LONG_MAX' is defined in header '<climits>'; did you forget to '#include <climits>'?
    scons: *** [platform/linuxbsd/display_server_x11.linuxbsd.tools.64.o] Error 1
    scons: building terminated because of errors.

--- a/platform/linuxbsd/display_server_x11.cpp
+++ b/platform/linuxbsd/display_server_x11.cpp
@@ -43,6 +43,7 @@
 #include "servers/rendering/rasterizer_rd/rasterizer_rd.h"
 #endif
 
+#include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>

